go/extractor/cli/go-autobuilder/go-autobuilder.go

```diff
@@ -719,82 +719,150 @@ func installDependenciesAndBuild() {
-			"). Writing an environment file not specifying any version of Go."
+// Assuming `v.goModVersionFound` is false, emit a diagnostic and return the version to install,
+// or the empty string if we should not attempt to install a version of Go.
+func getVersionWhenGoModVersionNotFound(v versionInfo) (msg, version string) {
+	if !v.goEnvVersionFound {
+		// We definitely need to install a version. We have no indication which version was
+		// intended to be used to build this project. Go versions are generally backwards
+		// compatible, so we install the maximum supported version.
+		msg = "No version of Go installed and no `go.mod` file found. Writing an environment " +
+			"file specifying the maximum supported version of Go (" + maxGoVersion + ")."
+		version = maxGoVersion
+		diagnostics.EmitNoGoModAndNoGoEnv(msg)
+	} else if outsideSupportedRange(v.goEnvVersion) {
+		// We definitely need to install a version. We have no indication which version was
+		// intended to be used to build this project. Go versions are generally backwards
+		// compatible, so we install the maximum supported version.
+		msg = "No `go.mod` file found. The version of Go installed in the environment (" +
+			v.goEnvVersion + ") is outside of the supported range (" + minGoVersion + "-" +
+			maxGoVersion + "). Writing an environment file specifying the maximum supported " +
+			"version of Go (" + maxGoVersion + ")."
+		version = maxGoVersion
+		diagnostics.EmitNoGoModAndGoEnvUnsupported(msg)
+	} else {
+		// The version of Go that is installed is supported. We have no indication which version
+		// was intended to be used to build this project. We assume that the installed version is
+		// suitable and do not install a version of Go.
+		msg = "No `go.mod` file found. Version " + v.goEnvVersion + " installed in the " +
+			"environment is supported. Writing an environment file not specifying any " +
+			"version of Go."
 		version = ""
-		diagnostics.EmitUnsupportedVersionEnvironment(msg)
+		diagnostics.EmitNoGoModAndGoEnvSupported(msg)
 	}
 
 	return msg, version
 }
 
-// Check if either `v.goEnvVersionFound` or `v.goModVersionFound` are false. If so, emit
-// a diagnostic and return the version to install, or the empty string if we should not attempt to
-// install a version of Go. We assume that `checkForUnsupportedVersions` has already been
-// called, so any versions that are found are within the supported range.
-func checkForVersionsNotFound(v versionInfo) (msg, version string) {
-	if !v.goEnvVersionFound && !v.goModVersionFound {
-		msg = "No version of Go installed and no `go.mod` file found. Writing an environment " +
-			"file specifying the maximum supported version of Go (" + maxGoVersion + ")."
-		version = maxGoVersion
-		diagnostics.EmitNoGoModAndNoGoEnv(msg)
-	}
+// Assuming `v.goModVersion` is above the supported range, emit a diagnostic and return the
+// version to install, or the empty string if we should not attempt to install a version of Go.
```

Comment:
This comment suggests that this function may return a version to install, which it never does.